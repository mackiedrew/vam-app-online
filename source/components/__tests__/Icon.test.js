import Subject from "../Icon";

const mockIcon = "not_an_icon";

describe("<Icon />", () => {
  it("renders without crashing with props", () => {
    shallow(<Subject icon={mockIcon} />);
  });

  describe("renders the correct elements", () => {
    const SubjectWithIcon = shallow(<Subject icon={mockIcon} />);
    it("<i className='material-icons'>", () => {
      expect(SubjectWithIcon.is("i.material-icons")).toEqual(true);
    });
  });

  describe("renders correctly", () => {
    it("with icon", () => {
      const tree = renderer.create(<Subject icon={mockIcon} />).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("with icon and size", () => {
      const tree = renderer
        .create(<Subject icon={mockIcon} size={50} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
