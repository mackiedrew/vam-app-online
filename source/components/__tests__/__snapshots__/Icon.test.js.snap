// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly with icon 1`] = `
<i
  className="material-icons icon"
  style={
    Object {
      "fontSize": "24px",
      "height": "24px",
      "width": "24px",
    }
  }
>
  not_an_icon
</i>
`;

exports[`<Icon /> renders correctly with icon and size 1`] = `
<i
  className="material-icons icon"
  style={
    Object {
      "fontSize": "50px",
      "height": "50px",
      "width": "50px",
    }
  }
>
  not_an_icon
</i>
`;
